app/components/CacheAnalysis.vue

```diff
@@ -35,8 +35,16 @@ const cacheAnalysis = computed(() =>
 )
 
 // Helper function to handle hover events on data keys
-const handleDataKeyHover = (dataKey: string, dataValue: string, rawValue: unknown = null) => {
-  setHover(dataKey, dataValue, rawValue)
+const handleDataKeyHover = (dataKey: string, rawValue: unknown) => {
+  // Type guard to ensure we have a valid value for display
+  if (typeof rawValue === 'boolean' || typeof rawValue === 'number' || typeof rawValue === 'string' || rawValue instanceof Date) {
+    const displayValue = getDisplayValue(rawValue)
+    setHover(dataKey, displayValue, rawValue)
+  } else {
+    // For unknown types, convert to string
+    const displayValue = String(rawValue)
+    setHover(dataKey, displayValue, rawValue)
+  }
```

Comment:
I think this can be simplified? We have TypeScript... We can just type the arg as the union of the expected types. We'll never call this with anything else.